.github/workflows/ci.yaml

```diff
@@ -0,0 +1,54 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+
+        ocaml-compiler:
+          - 4.04.x
+          - 4.05.x
+          - 4.06.x
+          - 4.07.x
+          - 4.08.x
+          - 4.09.x
+          - 4.10.x
+          - 4.11.x
+          - 4.12.x
+          - 4.13.x
```

Comment:
Is there a reason to list so many compilers? I am somewhat hesitant to add so many builds because:

1. If it works with the lowest compiler and the highest it *should* work with any in between (since this doesn't use any CPPO or version-specific magic)
2. So many builds take long and it sucks if some of these fails for a random reason (like network not working)

Reply:
I agree, I was making a guess of the oldest OCaml that might work.

Reply:
According to the Travis file it should be 4.04 and back in the day when Travis worked I believe it built on that version.

It seems to currently fail on ocaml-ci because `asn1-combinators` uses `Result` but doesn't have a dependency. I assume this is because rresult.0.7.0 dropped the `result` package and thus the transitive dependency disappeared, breaking `asn1-combinators`.

Reply:
@ki-ty-kate fixed [the issue in opam](https://github.com/ocaml/opam-repository/pull/20136), so this is good to go once squashed :+1: